### ISSUE TITLE
Regression in gcp, cant fetch metadata

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#440]
Originally reported [on the discussion forum](https://discussion.fedoraproject.org/t/regression-in-gcp-cant-fetch-metadata/18368).

[Recently](https://github.com/coreos/ignition-dracut/pull/159) we added a service to teardown networking in the initramfs before moving on to the real root. Now there is a regression showing up on at least on GCP where the bootup gets hung trying to reach the metadata service:

```
[[    8.948846] systemd[1]: Stopping Tear down initramfs networking...
[0;32m  OK  [0m] Stopped [0;1;39mdracut initqueue hook[0m.
[    8.953828] systemd[1]: dracut-pre-mount.service: Succeeded.
[[0;32m  OK  [0m] Stopped [0;1;39mIgnition OSTree: Grow root filesystem[0m.
[    8.959105] coreos-teardown-initramfs-network[718]: info: taking down network device: eth0
[[0;32m  OK  [0m] Stopped [0;1;39mIgnition OSTree: Mount (firstboot) /sysroot[0m.
[    8.965888] systemd[1]: Stopped dracut pre-mount hook.
[[0;32m  OK  [0m] Started [0;1;39mCleaning Up and Shutting Down Daemons[0m.
[    8.972818] coreos-teardown-initramfs-network[718]: RTNETLINK answers: Operation not supported
[    8.979384] systemd[1]: dracut-initqueue.service: Succeeded.
[    8.983691] coreos-teardown-initramfs-network[718]: info: taking down network device: lo
[    8.989777] systemd[1]: Stopped dracut initqueue hook.
[    8.992795] coreos-teardown-initramfs-network[718]: RTNETLINK answers: Operation not supported
[[0;32m  OK  [0[    8.997772] systemd[1]: ignition-ostree-growfs.service: Succeeded.
m] Stopped [0;1;39mTear down initramfs networki[    9.000059] coreos-teardown-initramfs-network[718]: info: flushing all routing
ng[0m.
[    9.001710] coreos-teardown-initramfs-network[718]: info: no networking config is defined in the real root
[    9.004213] coreos-teardown-initramfs-network[718]: info: propagating initramfs networking config to the real root
[[0;32m  OK  [0m] Stopped [0;1[    9.008263] systemd[1]: Stopped Ignition OSTree: Grow root filesystem.
;39mIgnition (files)[0m.
[[0;32m  OK  [0m] Stopped [0;1;39mPopulate OST[    9.012479] systemd[1]: ignition-ostree-mount-firstboot-sysroot.service: Succeeded.
ree /var[0m.
[    9.015100] systemd[1]: Stopped Ignition OSTree: Mount (firstboot) /sysroot.
[    9.017518] systemd[1]: initrd-cleanup.service: Succeeded.
         Stopping [0;1;39mIgnition (mount)[0m...
[    9.025880] systemd[1]: Started Cleaning Up and Shutting Down Daemons.
[    9.030986] systemd[1]: coreos-teardown-initramfs-network.service: Succeeded.
[    9.034202] ignition[734]: INFO     : Ignition 2.2.1
[    9.035775] ignition[734]: INFO     : Stage: umount
[    9.037341] ignition[734]: INFO     : reading system config file "/usr/lib/ignition/base.ign"
[    9.039768] ignition[734]: DEBUG    : parsing config with SHA512: ff6a5153be363997e4d5d3ea8cc4048373a457c48c4a5b134a08a30aacd167c1e0f099f0bdf1e24c99ad180628cd02b767b863b5fe3a8fce3fe1886847eb8e2e
[    9.043384] ignition[734]: DEBUG    : parsed url from cmdline: ""
[    9.044883] ignition[734]: INFO     : no config URL provided
[    9.046424] ignition[734]: INFO     : reading system config file "/usr/lib/ignition/user.ign"
[    9.048133] ignition[734]: INFO     : no config at "/usr/lib/ignition/user.ign"
[    9.050674] ignition[734]: INFO     : GET http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data: attempt #1
[    9.054325] ignition[734]: INFO     : GET error: Get http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data: dial tcp: lookup metadata.google.internal on 169.254.169.254:53: dial udp 169.254.169.254:53: connect: network is unreachable
[    9.061034] systemd[1]: Stopped Tear down initramfs networking.
[    9.062869] systemd[1]: ignition-files.service: Succeeded.
[    9.064317] systemd[1]: Stopped Ignition (files).
[    9.065587] systemd[1]: ignition-ostree-populate-var.service: Succeeded.
[    9.067495] systemd[1]: Stopped Populate OSTree /var.
[    9.068964] systemd[1]: Stopping Ignition (mount)...
[    9.235856] ignition[734]: INFO     : GET http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data: attempt #2
[    9.239128] ignition[734]: INFO     : GET error: Get http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data: dial tcp: lookup metadata.google.internal on 169.254.169.254:53: dial udp 169.254.169.254:53: connect: network is unreachable
[    9.636829] ignition[734]: INFO     : GET http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data: attempt #3
[    9.640308] ignition[734]: INFO     : GET error: Get http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data: dial tcp: lookup metadata.google.internal on 169.254.169.254:53: dial udp 169.254.169.254:53: connect: network is unreachable
```

This appears to be because of two reasons:

- the ExecStop from coreos-teardown-initramfs-network.service is racing with the ExecStop from ignition-mount.service
- ignition apparently retries to fetch metadata each time if it received an empty config: https://github.com/coreos/ignition/issues/950

**Comment by dustymabe** [comment]
possible fix in  https://github.com/coreos/ignition-dracut/pull/163

**Comment by bgilbert** [comment]
For the reason mentioned in https://github.com/coreos/ignition-dracut/pull/163#issuecomment-605749299, I wonder if it'd be better to apply https://github.com/coreos/ignition/pull/952 and fix https://github.com/coreos/ignition/issues/950.

**Comment by dustymabe** [comment]
> For the reason mentioned in [coreos/ignition-dracut#163 (comment)](https://github.com/coreos/ignition-dracut/pull/163#issuecomment-605749299), I wonder if it'd be better to apply [coreos/ignition#952](https://github.com/coreos/ignition/pull/952) and fix [coreos/ignition#950](https://github.com/coreos/ignition/issues/950).

sounds good to me but we'd need to do that ASAP so we can update our testing branch and get some soak time before next week's releases

**Comment by dustymabe** [comment]
this should be fixed in the latest testing release[`31.20200323.2.1`](https://github.com/coreos/fedora-coreos-streams/issues/76)

**Comment by dustymabe** [comment]
The fix for this went into stable stream release `31.20200323.3.2`.
